components/domains/folder/FolderButtonList.tsx

```diff
@@ -0,0 +1,112 @@
+import styled from "styled-components";
+import React from "react";
+import { getSelectData } from "../../../pages/api/FolderApi";
+import { Folder } from "../../../types/folder";
+import { Link } from "../../../types/common";
+
+function Button({
+  folder,
+  setSelectFolderLinks,
+  setId,
+}: {
+  folder: Folder;
+  setSelectFolderLinks: React.Dispatch<React.SetStateAction<Link[]>>;
+  setId: React.Dispatch<React.SetStateAction<number>>;
+}) {
+  const handleChangeID = async () => {
+    const { data } = await getSelectData(folder.id);
+    setSelectFolderLinks(data);
+    setId(folder.id);
+  };
+
+  return (
+    <StyledButton
+      onClick={() => {
+        handleChangeID();
+      }}
+    >
+      {folder && folder.name}
+    </StyledButton>
+  );
+}
+
+function FolderButtonList({
+  folderList,
+  setSelectFolderLinks,
+  setId,
+  openModal,
+}: {
+  folderList: Folder[];
+  setSelectFolderLinks: React.Dispatch<React.SetStateAction<Link[]>>;
+  setId: React.Dispatch<React.SetStateAction<number>>;
+  openModal: () => void;
+}) {
+  return (
+    <div>
+      <StyledButtonBox>
+        <StyledButtons>
+          {folderList?.map((folder) => (
+            <Button
+              folder={folder}
+              key={folder.id}
+              setSelectFolderLinks={setSelectFolderLinks}
+              setId={setId}
+            />
+          ))}
+        </StyledButtons>
+        <StyledAddLink
+          onClick={() => {
+            openModal("폴더추가");
+          }}
+        >
+          +
+        </StyledAddLink>
+      </StyledButtonBox>
+    </div>
+  );
+}
+
+const StyledButton = styled.button`
+  background-color: #ffffff;
+  border: 1px solid var(--primary-color);
+  font-size: 1.6rem;
+  border-radius: 0.5rem;
+  padding: 0.8rem 1.2rem;
+  cursor: pointer;
+
+  &:hover {
+    background-color: var(--primary-color);
+    color: #ffffff;
+  }
+
+  /* Mobile*/
+  @media (max-width: 390px) {
+    padding: 0.6rem 1.2rem;
+  }
+`;
+
+const StyledButtonBox = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 2.4rem;
+`;
+
+const StyledButtons = styled.div`
+  display: flex;
+  justify-content: flex-start;
+  gap: 1.2rem;
+  flex-wrap: wrap;
+`;
+
+const StyledAddLink = styled.div`
```

Comment:
어차피 재사용될 가능성이 낮은, Folder 추가만을 위한 스타일의 버튼 같아보여요.
컴포넌트를 만들 때, 재사용가능성을 염두하는 건 좋지만, 대부분은 특정 도메인에만 사용됩니다.
이름도 그것을 드러내서 정확하면 좋겠습니다. 그리고 진짜 Link라면 `div` 면 안되고 `a`인게 자연스러울거같아요.

```suggestion
const AddFolderLink = styled.a`
```